### PR TITLE
#889 On OracleDB, primary key is considered missing if index_name != constraint_name

### PR DESCRIPTION
**Pull request by marius-meissner**
Same as pull request #889, but with added test.
Fixes issue doctrine/dbal#1234 and doctrine/doctrine2#6206

### CODE REVIEW

File: tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php [hunk 1]
```diff
@@ -233,6 +233,29 @@ public function testListTableColumnsSameTableNamesInDifferentSchemas()
     }
 
     /**
+     * @group DBAL-1234
+     */
+    public function testListTableIndexesPrimaryKeyConstraintNameDiffersFromIndexName()
+    {
+        $table = new Table('list_table_indexes_pk_id_test');
+        $table->setSchemaConfig($this->_sm->createSchemaConfig());
+        $table->addColumn('id', 'integer', ['notnull' => true]);
```
Comment (deeky666) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php]: Please do not use short array syntax here, as we target this patch for `2.5` where we still support PHP `5.3`.

**Review by deeky666 — DISMISSED** [review]

File: tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php (hunk 1 above)
Comment (Ocramius) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php]: @deeky666 let's fix it ourselves during backport

Comment (SkydiveMarius) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php]: @deeky666: Thank you for the hint. I just replaced all shorthand array syntax.

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
:ship: thanks @SkydiveMarius!

**Comment by Ocramius** [comment]
Backported to `2.5` via doctrine/dbal@76b65d73ec75b4e1dc672731d0e7be1437870abd

**Pull request merged**